shell/platform/windows/text_input_plugin.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/text_input_plugin.h"
+
+#include <windows.h>
+#include <cstdint>
+#include <iostream>
+
+#include "flutter/shell/platform/common/cpp/client_wrapper/include/flutter/json_method_codec.h"
+
+static constexpr char kSetEditingStateMethod[] = "TextInput.setEditingState";
+static constexpr char kClearClientMethod[] = "TextInput.clearClient";
+static constexpr char kSetClientMethod[] = "TextInput.setClient";
+static constexpr char kShowMethod[] = "TextInput.show";
+static constexpr char kHideMethod[] = "TextInput.hide";
+
+static constexpr char kMultilineInputType[] = "TextInputType.multiline";
+
+static constexpr char kUpdateEditingStateMethod[] =
+    "TextInputClient.updateEditingState";
+static constexpr char kPerformActionMethod[] = "TextInputClient.performAction";
+
+static constexpr char kSelectionBaseKey[] = "selectionBase";
+static constexpr char kSelectionExtentKey[] = "selectionExtent";
+
+static constexpr char kTextKey[] = "text";
+
+static constexpr char kChannelName[] = "flutter/textinput";
+
+static constexpr char kBadArgumentError[] = "Bad Arguments";
+static constexpr char kInternalConsistencyError[] =
+    "Internal Consistency Error";
+
+static constexpr uint32_t kInputModelLimit = 256;
+
+namespace flutter {
+
+void TextInputPlugin::CharHook(FlutterWindow* window, unsigned int code_point) {
+  if (active_model_ == nullptr) {
+    return;
+  }
+  // TODO(awdavies): Actually handle potential unicode characters. Probably
```

Comment:
Update TODO; awdavies would not be implementing this. It's fine to use a bug reference instead of a person (In this case 30661)